src/utils/helperFunctions/metadataFilters.ts

```diff
@@ -1,43 +1,33 @@
 import { DataStreamWithRoutes } from "src/utils/api/dataStreams";
 
-export interface getDataStreamIdsParams {
-  data: DataStreamWithRoutes[];
-  writeOnly?: boolean;
-}
-
-export const getDataStreamIds = ({
-  data,
-  writeOnly = false,
-}: getDataStreamIdsParams): string[] => {
-  return data
-    .filter(
-      (ds) => !writeOnly || ds.routes.some((route) => route.writePermissions)
-    )
-    .map((ds) => ds.datastream.name);
+export const getWriteOnlyDatastreamsWithRoutes = (
+  data: DataStreamWithRoutes[]
+): DataStreamWithRoutes[] => {
+  const writeOnlyDatastreamsWithRoutes = data
+    .filter((ds) => ds.routes.some((route) => route.writePermissions))
```

Comment:
This is fine for now, but at some point I think this filtering should be handled by the backend.  Could easily add a query param on the fetch for data streams.

Reply:
These are coming from B2c token, so maybe changing token loader to store two versions.